### DISCUSSION TITLE
Is there an issue or discussion tracking the tasks/issues until stability is reached?

### DISCUSSION THREAD

**Discussion by coltenkrauter** [immich-app/immich#10867] [General]
Is there an issue or discussion tracking the tasks/issues until stability is reached?

It would be nice to understand the priorities of the developers and maintainers of this project and be able to gauge the amount of work until immich is considered stable.

I would imagine that such an issue exists, sort of an epic, but I didn't see it. 

Additionally, it would be neat to have such an issue linked in the README.

**Comment by mertalev** [comment]
We have a [public roadmap](https://immich.app/roadmap/) with some of the high level goals we have for the stable release, but I don't think we have a tracking issue or similar for this or the more technical goals. 

Generally, we tend to use issues for bugs and have features as either GitHub discussions, draft PRs or Discord focus discussions. But this isn't a hard and fast rule, just our typical workflow.

**Reply by coltenkrauter** [reply]
Would you be open to creating such a tracking issue?

Thanks for the roadmap link.

**Reply by mertalev** [reply]
I'm personally open to it, but I'd have to discuss with the other maintainers to get their thoughts.

**Reply by bo0tzz** [reply]
We don't have a specific set of things to do before stable. The public roadmap is the main feature overview, but there are also many refactors, bugfixes etc, many of which currently only live in the maintainers heads. Personally I don't think it's worth the effort right now to write that all down.

**Reply by btajuddin** [reply]
The public roadmap just says sometime this year. We're 2/3 of the way through the year, and it doesn't instill a lot of trust when the project cannot provide an idea of what is required to reach that milestone. Immich has gone through multiple breaking changes this year that have created a burden on operators, especially since the mobile app also seems to move forward without much of a mind to backward compatibility.

Without some insight into the path to stability for Immich, I'm much more likely to look elsewhere at this point. I know this project has always been advertised as under active development, and I appreciate the effort put into it. I also don't care about most of the features released this year that forced me to do extra work to upgrade. I would prefer a stable system over the scope creep that we're seeing (at least, that's what it looks like from the outside).

Again, I appreciate all you do for the project and the community. I just want to encourage you toward more transparency. I've seen too many good projects die for lack of it.

**Reply by alextran1502** [reply]
@btajuddin, I understand your concerns. I believe we have been doing a pretty good job of managing expectations. So people know that a stable release will sometimes be at the end of this year or early next year. I don't want to make detailed updates on everything we do since it will take time away from actual programming work. We have a small team.

<img width="936" alt="image" src="https://github.com/user-attachments/assets/4b4b4490-875e-40fd-bf45-2da7f410e61b">

From the roadmap items, those are the things we want to hit before having a stable release; most of the items are being worked on, and even more so, to handle backward compatibility. You can always follow the commits and PR changes for more details. 

As a software developer veteran, I believe you know estimation in software development is often subjective. We set goals so we can strive for our best. If you have ideas of how to better our communication for the project and open source software, please let me know.
